plugins/integrations/kubernetes-service/src/main/java/com/cloud/kubernetescluster/KubernetesClusterManagerImpl.java

```diff
@@ -0,0 +1,3061 @@
+            LOGGER.warn(msg, e);
+            stateTransitTo(kubernetesCluster.getId(), KubernetesCluster.Event.CreateFailed);
+            throw new ManagementServerException(msg, e);
+        }
+
+        // Port forwarding rule fo SSH access on each node VM
+        final long publicIpId = publicIp.getId();
+        final long networkId = kubernetesCluster.getNetworkId();
+        final long accountId = account.getId();
+        final long domainId = account.getDomainId();
+
+        for (int i = 0; i < clusterVMIds.size(); ++i) {
+            long vmId = clusterVMIds.get(i);
+            Nic vmNic = networkModel.getNicInNetwork(vmId, kubernetesCluster.getNetworkId());
+            final Ip vmIp = new Ip(vmNic.getIPv4Address());
+            final long vmIdFinal = vmId;
+            final int srcPortFinal = CLUSTER_NODES_DEFAULT_START_SSH_PORT + i;
+            try {
+                PortForwardingRuleVO pfRule = Transaction.execute(new TransactionCallbackWithException<PortForwardingRuleVO, NetworkRuleConflictException>() {
+                    @Override
+                    public PortForwardingRuleVO doInTransaction(TransactionStatus status) throws NetworkRuleConflictException {
+                        PortForwardingRuleVO newRule =
+                                new PortForwardingRuleVO(null, publicIpId,
+                                        srcPortFinal, srcPortFinal,
+                                        vmIp,
+                                        22, 22,
+                                        "tcp", networkId, accountId, domainId, vmIdFinal);
+                        newRule.setDisplay(true);
+                        newRule.setState(FirewallRule.State.Add);
+                        newRule = portForwardingRulesDao.persist(newRule);
+                        return newRule;
+                    }
+                });
+                rulesService.applyPortForwardingRules(publicIp.getId(), account);
+                LOGGER.debug(String.format("Provisioned SSH port forwarding rule from port %d to 22 on %s to the VM IP: %s in Kubernetes cluster ID: %s", srcPortFinal, publicIp.getAddress().addr(), vmIp, kubernetesCluster.getUuid()));
+            } catch (Exception e) {
+                String msg = String.format("Failed to activate SSH port forwarding rules for the Kubernetes cluster ID: %s", kubernetesCluster.getUuid());
+                LOGGER.warn(msg, e);
+                stateTransitTo(kubernetesCluster.getId(), KubernetesCluster.Event.CreateFailed);
+                throw new ManagementServerException(msg, e);
+            }
+        }
+    }
+
+    // Open up  firewall ports NODES_DEFAULT_START_SSH_PORT to NODES_DEFAULT_START_SSH_PORT+n for SSH access. Also create port-forwarding
+    // rule to forward public IP traffic to all node VM private IP. Existing node VMs before scaling
+    // will already be having these rules
```

Comment:
another big method, these big methods all warrant their own helper/worker class.

Reply:
I saw you created workers for these big methods, why are these still so big?